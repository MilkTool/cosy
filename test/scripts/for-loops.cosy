fun f() {
    for j in 0..2 {
        print "loop #" + j
    }
    var from = 2
    var to = 4
    for j in from..to {
        print "loop #" + j
    }
}

fun f1() {
    mut i = 0
    for i < 5 {
        print "conditional loop #" + i
        i = i + 1
        if (i >= 3) return _
    }
}

fun f2() {
    mut j = 0
    for {
        print "infinite loop #" + j
        j = j + 1
        if (j >= 3) return _
    }
}

f()
f1()
f2()
