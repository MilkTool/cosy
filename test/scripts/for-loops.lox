fun f() {
    for j in 0..3 {
        print "loop #" + j
    }
}

fun f1() {
    mut i = 0
    for i < 5 {
        print "conditional loop #" + i
        i = i + 1
        if (i >= 3) return _
    }
}

fun f2() {
    mut j = 0
    for {
        print "infinite loop #" + j
        j = j + 1
        if (j >= 3) return _
    }
}

f()
f1()
f2()
