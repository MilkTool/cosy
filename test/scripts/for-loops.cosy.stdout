loop
loop
loop #0
loop #1
loop #2
loop #3
conditional loop #0
conditional loop #1
conditional loop #2
infinite loop #0
infinite loop #1
infinite loop #2
